code/game/objects/items/motion_detector.dm

```diff
@@ -1,179 +1,127 @@
-
-#define MOTION_DETECTOR_LONG 0
-#define MOTION_DETECTOR_SHORT 1
-
-#define MOTION_DETECTOR_HOSTILE ""
-#define MOTION_DETECTOR_FRIENDLY "_friendly"
-#define MOTION_DETECTOR_DEAD "_dead"
-#define MOTION_DETECTOR_FUBAR "_fubar" //i.e. can't be revived. Might have useful gear to loot though!
-
-
 /obj/effect/detector_blip
 	icon = 'icons/Marine/marine-items.dmi'
-	icon_state = "detector_blip"
+	icon_state = "detector_blip_dir"
 	layer = BELOW_FULLSCREEN_LAYER
+	plane = ABOVE_HUD_PLANE
 	var/identifier = MOTION_DETECTOR_HOSTILE
 	var/edge_blip = FALSE
-	var/turf/center
-	var/turf/target_turf
+	///The image shown on the user screen
+	var/image/blip_image
+
+/// Print the blip on the operator screen
+/obj/effect/detector_blip/proc/show_bip(mob/operator, mob/target, screen_pos_x, screen_pos_y,)
+	if(!operator?.client)
+		return
+	if(edge_blip) // we want to draw the blip directly on the screen edges, so we use client.screen
+		moveToNullspace()
+		icon_state = initial(icon_state)
+		screen_loc = "[screen_pos_x],[screen_pos_y]"
+		operator.client.screen += src
+	else //we want to draw the blip on the target, so we need a game loc, so we use client.images
+		forceMove(get_turf(target))
+		icon_state = ""
+		if(!blip_image)
+			blip_image = image('icons/Marine/marine-items.dmi', src, "detector_blip[identifier]")
+		blip_image.layer = BELOW_FULLSCREEN_LAYER
+		operator.client.images |= blip_image
```

Comment:
Fight me tivi

Reply:
Yes I will this is garbage and yu know it